### PR TITLE
cli: encode/decode SpawnApp tx

### PR DESCRIPTION
**Pull request by moshababo**
Closing https://github.com/spacemeshos/svm/issues/77.

WIP: docs, tests (cli, `WasmValue::try_from(&str)`)

### CODE REVIEW

File: crates/svm-app/src/types/wasm_value.rs [hunk 1]
```diff
@@ -7,3 +11,47 @@ pub enum WasmValue {
     /// A 64-bit integer.
     I64(u64),
 }
+
```
Comment (YaronWittenstein) [inline crates/svm-app/src/types/wasm_value.rs]: I'm ambivalent about whether this is the right place for this piece.
I know it's targeted for the SVM CLI but besides that, I don't see any useful scenario for this.

I'd rather move it to the `svm-cli` crate.

File: crates/svm-cli/src/cli.rs [hunk 2]
```diff
@@ -13,21 +12,21 @@ pub fn new_app<'a, 'b>() -> App<'a, 'b> {
                 .about("Encode various transaction types")
                 .setting(AppSettings::SubcommandRequiredElseHelp)
                 .subcommand(
-                    App::new("app-template")
```
Comment (YaronWittenstein) [inline crates/svm-cli/src/cli.rs]: I personally love using `_` instead of `_` but it's OK with both.

File: crates/svm-cli/src/spawn_app.rs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            let e = format!(
+                "failed to decode hex string '{}': {}",
+                ctor_buf_hex,
+                e.to_string()
+            );
+            return Err(e.into());
+        }
+    };
+
+    let mut ctor_args_vals: Option<Vec<WasmValue>> = None;
```
Comment (YaronWittenstein) [inline crates/svm-cli/src/spawn_app.rs]: please rename it as `ctor_args`.

File: crates/svm-cli/src/spawn_app.rs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    if let Some(args) = ctor_args {
+        let mut vals: Vec<WasmValue> = Vec::new();
```
Comment (YaronWittenstein) [inline crates/svm-cli/src/spawn_app.rs]: the `Vec<WasmValue>` seems redundant.

File: crates/svm-cli/src/spawn_app.rs [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        let mut errs: Vec<String> = Vec::new();
```
Comment (YaronWittenstein) [inline crates/svm-cli/src/spawn_app.rs]: the `Vec<String>` seems redundant.

File: crates/svm-cli/src/spawn_app.rs [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+        ctor_args_vals = Some(vals);
+    }
+
+    let mut builder = SpawnAppBuilder::new()
+        .with_version(version)
+        .with_template(&template_addr.into())
+        .with_ctor_index(ctor_idx)
+        .with_ctor_buf(&ctor_buf);
+    if let Some(vals) = ctor_args_vals {
```
Comment (YaronWittenstein) [inline crates/svm-cli/src/spawn_app.rs]: can be written as:
`if let Some(ctor_args) = ctor_args { ... }`

File: crates/svm-app/src/types/wasm_value.rs [hunk 7]
```diff
@@ -7,3 +11,47 @@ pub enum WasmValue {
...
+            return Err(InvalidFormat);
+        }
+        let mid = len - TYPE_CHARS_LEN;
+        let num: String = value.chars().take(mid).collect();
+        let num_t: String = value.chars().skip(mid).collect();
+
+        match &num_t[..] {
+            "i32" => {
+                match u32::from_str(&num) {
+                    Ok(v) => return Ok(WasmValue::I32(v)),
```
Comment (YaronWittenstein) [inline crates/svm-app/src/types/wasm_value.rs]: the `return` seems redundant.

File: crates/svm-cli/src/cli.rs (hunk 2 above)
Comment (moshababo) [reply crates/svm-cli/src/cli.rs]: I didn't understand what you mean.. you mentioned the same char twice.

File: crates/svm-cli/src/spawn_app.rs (hunk 3 above)
Comment (moshababo) [reply crates/svm-cli/src/spawn_app.rs]: A different name is needed since i'm using `Vec<&str>` to populate `Vec<WasmValue>`.

**Pull request closed without merging**